deployment/cdk/opensearch-service-migration/lib/service-stacks/migration-analytics-stack.ts

```diff
@@ -0,0 +1,102 @@
+import {StackPropsExt} from "../stack-composer";
+import {
+  BastionHostLinux,
+  BlockDeviceVolume,
+  MachineImage,
+  Port,
+  SecurityGroup,
+  IVpc,
+} from "aws-cdk-lib/aws-ec2";
+import {PortMapping, Protocol, ServiceConnectService} from "aws-cdk-lib/aws-ecs";
+import {Construct} from "constructs";
+import {join} from "path";
+import {MigrationServiceCore} from "./migration-service-core";
+import {StringParameter} from "aws-cdk-lib/aws-ssm";
+
+export interface MigrationAnalyticsProps extends StackPropsExt {
+    readonly vpc: IVpc,
+}
+
+// The MigrationAnalyticsStack consists of the OpenTelemetry Collector ECS container & an
+// Bastion host to allow access to the opensearch dashboard.
+export class MigrationAnalyticsStack extends MigrationServiceCore {
+
+    constructor(scope: Construct, id: string, props: MigrationAnalyticsProps) {
+        super(scope, id, props)
+
+        // Bastion Security Group
+        const bastionSecurityGroup = new SecurityGroup(
+          this,
+          "analyticsDashboardBastionSecurityGroup",
+          {
+            vpc: props.vpc,
+            allowAllOutbound: true,
+            securityGroupName: "analyticsDashboardBastionSecurityGroup",
+          }
+        );
+        let securityGroups = [
+            SecurityGroup.fromSecurityGroupId(this, "serviceConnectSG", StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/serviceConnectSecurityGroupId`)),
+            SecurityGroup.fromSecurityGroupId(this, "migrationAnalyticsSGId", StringParameter.valueForStringParameter(this, `/migration/${props.stage}/${props.defaultDeployId}/analyticsDomainSGId`)),
+            bastionSecurityGroup
+        ]
+        securityGroups[1].addIngressRule(bastionSecurityGroup, Port.tcp(443))
```

Comment:
This makes me a little nervous to alter the resource of another stack. It seems like longer term we may want to create the bastion security group with a flag in the network stack and then apply it to the analytics domain as an additional security group as well. What are your thoughts?